tests/it/queue.rs

```diff
@@ -5,34 +5,32 @@ use broker::queue::{self, Queue, Receiver, Sender};
 async fn racing_senders_err() {
     let _root = set_temp_data_root();
 
-    let _first = Sender::open(Queue::Echo).await.expect("must open first");
-    let err = Sender::open(Queue::Echo)
-        .await
-        .expect_err("must fail to open second");
-    assert_error_stack_snapshot!(&"sender", err);
+    let _first: Sender<Vec<u8>> = Sender::open(Queue::Echo).await.expect("must open first");
+    let second: Result<Sender<Vec<u8>>, _> = Sender::open(Queue::Echo).await;
+    assert_error_stack_snapshot!(&"sender", second.expect_err("must fail to open second"));
 }
 
 #[tokio::test]
 async fn racing_receivers_err() {
     let _root = set_temp_data_root();
 
-    let _first = Receiver::open(Queue::Echo).await.expect("must open first");
-    let err = Receiver::open(Queue::Echo)
-        .await
-        .expect_err("must fail to open second");
-    assert_error_stack_snapshot!(&"receiver", err);
+    let _first: Receiver<Vec<u8>> = Receiver::open(Queue::Echo).await.expect("must open first");
+    let second: Result<Receiver<Vec<u8>>, _> = Receiver::open(Queue::Echo).await;
```

Comment:
This test is pretty consistently failing for me.

```
---- queue::racing_receivers_err stdout ----
thread 'queue::racing_receivers_err' panicked at 'must fail to open second: Receiver([OPAQUE yaque::Receiver])', tests/it/queue.rs:19:22
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    queue::racing_receivers_err

```

Every once in a while, I see this test fail instead:

```
failures:

---- queue::echo stdout ----
thread 'queue::echo' panicked at 'must open queue: open queue
├╴at /Users/scott/fossa/broker/src/queue.rs:134:14
├╴help: This may be caused by an underlying filesystem error, or the queue may already be open for sending.
│ If you are certain no other Broker instances are running, deleting the lock file may recover this error.
├╴context: Queue working state is stored on disk, and relies on a lockfile to guard access.
│ For this particular queue, this lock file is located at "/var/folders/45/k76d2pgj6mlb1vpx_2gyl7z80000gn/T/.tmp9op3vE/queue/Echo/send.lock".
│
├─▶ IO layer error
│   ╰╴at /Users/scott/fossa/broker/src/ext/io.rs:134:10
│
╰─▶ queue `/var/folders/45/k76d2pgj6mlb1vpx_2gyl7z80000gn/T/.tmp9op3vE/queue/Echo` sender side already in use
    ╰╴at /Users/scott/fossa/broker/src/ext/io.rs:147:38', tests/it/queue.rs:29:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

If I run just the "racing_receiver_err" test, then it passes (well, insta is complaining about a difference in colour codes, but other than that it passes).

I think ENV vars are stomping on each other in `set_temp_data_root`. Not sure if we can or should fix this now, but wanted to point it out

Reply:
Yeah, I am fixing that as part of this PR

Reply:
(not done yet though)